### ISSUE TITLE
New Rule: consistent-catch-param-name

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#40]
## Research Evidence

From vibecodingconsultant.com analysis of LLM patterns: LLMs use inconsistent variable naming in the same codebase (`e`, `err`, `error`, `ex`).

## Description

Enforce consistent naming for catch clause parameters across the codebase. LLMs frequently mix naming conventions in the same file.

## Options

```ts
{
  "llm-core/consistent-catch-param-name": ["error", {
    "name": "error" // or "e", "err", "ex", etc.
  }]
}
```

### Bad (inconsistent in same file)
```ts
try { ... } catch (e) { ... }
try { ... } catch (err) { ... }
try { ... } catch (error) { ... }
```

### Good
```ts
try { ... } catch (error) { ... }
try { ... } catch (error) { ... }
```

## Why This Matters for LLMs
- Consistent naming makes error handling patterns more recognizable
- Reduces cognitive load when reviewing AI-generated code
- Makes search/refactoring easier

## Priority
Medium - code quality improvement
